src/uct/base/uct_iface.h

```diff
@@ -353,25 +354,72 @@ typedef void (*uct_iface_mpool_init_obj_cb_t)(uct_iface_h iface, void *obj,
 
 
 /**
- * Base structure for private data held inside a pending request. Contains
- * a queue element so we can put this on a queue.
+ * Base structure for private data held inside a pending request for TLs
+ * which use ucs_arbiter_t to progress pending requests.
  */
 typedef struct {
-    ucs_queue_elem_t  queue;
-} uct_pending_req_priv_t;
+    ucs_arbiter_elem_t  arb_elem;
+} uct_pending_req_priv_arb_base_t;
 
 
+static UCS_F_ALWAYS_INLINE uct_pending_req_priv_arb_base_t *
+uct_pending_req_priv_arb_base(uct_pending_req_t* req)
```

Comment:
looks like this function is not used directly, maybe unite it with next function?

Reply:
These functions were added to suppress compilation error like this: http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/5497/label=ARM_test_node,worker=0/console
will restore

Reply:
isn't there other way to suppress them? like attribute(may_alias)?

Reply:
attribute in API type definition? not sure if it's a good idea

Reply:
can we somehow define it not in API?

Reply:
will try find something, but we cast here API defined `req->priv`...

Reply:
reworked by local pointer on stack